great_expectations/cli/toolkit.py

```diff
@@ -119,7 +127,7 @@ def get_or_create_expectation_suite(
 
 
 def get_default_expectation_suite_name(
-    data_asset_name: str,
+    data_asset_name: Union[str, None],
     batch_request: Optional[
         Union[str, Dict[str, Union[str, int, Dict[str, Any]]]]
```

Comment:
I don't understand why we landed on this possibly being a string. It would raise an error if it was, so maybe we should just proceed assuming it can't be.

Reply:
I was thinking there is an implicit assumption that this is only a `str` if`data_asset_name` is passed in, in which case it doesn't hit the `elif batch_request` conditional (which would have caused an error). 

Ideally we would encode this assumption by declaring `overload` functions that say...
>if data_asset is None, batch_request can be a string
>if data_asset is not None, batch_request must be a dict

But I was having some minor problems getting it to work as intended and I wasn't even sure if my assumptions there were correct.

https://mypy.readthedocs.io/en/stable/more_types.html#function-overloading